pkg/extension/sumologicextension/README.md

```diff
@@ -2,14 +2,57 @@
 
 **This extension is experimental and may receive breaking changes at any time.**
 
+This extension is to be used as part of Sumo Logic collector in conjuction with
+[`sumologicexporter`][sumologicexporter] in order to export telemetry data to
+[Sumo Logic][sumologic].
+
+It manages:
+
+* authentication (passing the provided credentials to `sumologicexporter`
+  when configured as extension in the same service)
+* registration (storing the registration info locally after successful registration
+  for later use)
+* heartbeats
+
+[sumologicexporter]: ../../exporter/sumologicexporter/
+[sumologic]: https://www.sumologic.com/
+
+## Implementation
+
+It implements [`HTTPClientAuthenticator`][httpclientauthenticator]
+and can be used as an authenticator for the
+[`configauth.Authentication`][configauth_authentication] option for HTTP clients.
+
+[httpclientauthenticator]: https://github.com/open-telemetry/opentelemetry-collector/blob/2e84285efc665798d76773b9901727e8836e9d8f/config/configauth/clientauth.go#L34-L39
+[configauth_authentication]: https://github.com/open-telemetry/opentelemetry-collector/blob/3f5c7180c51ed67a6f54158ede5e523822e9659e/config/configauth/configauth.go#L29-L33
+
+## Configuration
+
+* `access_id`: (required) access ID for Sumo Logic service, see
+  [help][credentials_help] for more details
+* `access_key`: (required) access key for Sumo Logic service, see
+  [help][credentials_help] for more details
+* `collector_name`: (required) name that will be used for locally stored
```

Comment:
Why is that required?

Reply:
It's required by the API as of now. We can tweak how it's being done by assigning a default collector name but I suggest we merge this README as is and only then change it when we change the behavior